source/common/upstream/cluster_manager_impl.h

```diff
@@ -523,6 +525,8 @@ class ClusterManagerImpl : public ClusterManager,
                                               LoadBalancerContext* context) override;
       Host::CreateConnectionData tcpConn(LoadBalancerContext* context) override;
       Http::AsyncClient& httpAsyncClient() override;
+      Tcp::AsyncTcpClientPtr tcpAsyncClient(LoadBalancerContext* context,
```

Comment:
The AsyncTcpClient has different semantics from the HTTP async client so it might be a bit confusing. While the HTTP async client allows you to start multiple stream/request to the cluster, the AsyncTcpClient is bound to 1 connection and not reusable.

Reply:
Right, this is a factory method of the AsyncTcpClient, do you have some other suggestions?.

Reply:
That depends on how this is going to be used, you might want follow the same pattern of the HTTP async client that starting multiple streams (connections in TCP).

Reply:
I see, 1 connection works for our use case right now, and I am thinking the client factory can be configurable e.g., we may add multiple streams support either in the current object or a new implementation of in future. WDYT? Thanks!

Reply:
SGTM, what I'm saying is that the name convention here confuses a bit compared to the http client above. The benefit of the http client is that you don't need to store/pass around a ClusterManager. In your use case, what if you want to retry that 1 connection?

Reply:
Thanks! We don't retry right now, I added an options to the factory, and then user can decide how the client is created & used.